frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx

```diff
@@ -235,7 +235,20 @@ class EmbedModalContent extends Component {
   }
 }
 
-export default connect(mapStateToProps)(EmbedModalContent);
+function getDefaultEmbeddingParams(props) {
+  const { resource, resourceParameters } = props;
+
+  return filterValidResourceParameters(
+    resource.embedding_params || {},
+    resourceParameters,
+  );
+}
+
+function filterValidResourceParameters(embeddingParams, resourceParameters) {
```

Comment:
nit: not sure if adding a separate function increases readability here

Reply:
This addresses the issue that the setup step keeps repeating the same options for `renderWithProvider` https://github.com/metabase/metabase/pull/27179#discussion_r1047530375